base/socket.jl

```diff
@@ -245,6 +245,7 @@ type InetAddr
     end
 end
 
+Base.call{T<:IPAddr}(::Type{InetAddr}, ip::T, port) = InetAddr{T}(ip, port)
```

Comment:
We don't have `typeof` inside type parameters very often do we? I would think the `InetAddr{T<:IPAddr}(ip::T, port) = InetAddr{T}(ip, port)` is a bit more idiomatic than `InetAddr(ip::IPAddr, port) = InetAddr{typeof(ip)}(ip, port)`